### PR TITLE
Theme add todo

### PR DESCRIPTION
**Pull request by shashikantmaurya0007**
feature:1.added theme and localstorage for that
            2.context to mangae todo
            3.modal to add todo
            4.list down todo and added localstorage for that

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { useTheme } from "../state/stateIndex";
+import { RiMoonClearFill } from "react-icons/ri";
+import { BsFillSunFill } from "react-icons/bs";
```
Comment (poojavishnoi) [inline src/components/Header.js:4]: you can also export these icons from componentIndex.js

File: src/styles/Header.module.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    min-height: 100vh;
+}
+
+.dark {
+    background-color: var(--dark_bg_color);
+    color: whitesmoke;
+}
+
+.sun {
+    color: yellow;
```
Comment (poojavishnoi) [inline src/styles/Header.module.css:36]: here also you can use variable

**Review by poojavishnoi — APPROVED** [review]
left few suggestions.Other than that everything is good

File: src/components/DisplayTodos.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      {todoState.map((ele, index) => {
+        console.log(ele.title, index);
+        return (
+          <div
+            className={`${styles.individual_todo} ${
+              theme === "dark" ? styles.dark : styles.light
+            }`}
+          >
+            <p>
+              {ele.title.trim().length > 20
```
Comment (SukanyaSen530) [inline src/components/DisplayTodos.jsx:21]: Suggestion - You can extract this logic into a function and return the output to show here.

**Review by SukanyaSen530 — APPROVED** [review]
Rest look's good :)

**Pull request merged**